project_task_stage_allow_timesheet/models/account_analytic_line.py

```diff
@@ -2,35 +2,36 @@
 # Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
-from odoo import api, fields, models, _
+from odoo import _, api, fields, models
 from odoo.exceptions import ValidationError
 
 
 class AccountAnalyticLine(models.Model):
 
-    _inherit = 'account.analytic.line'
+    _inherit = "account.analytic.line"
 
     task_id = fields.Many2one(
         domain=lambda self: self._get_task_domain(),
     )
 
-    @api.constrains('task_id')
+    @api.constrains("task_id")
     def _check_task_allow_timesheet(self):
         for rec in self:
             task = rec.task_id
             stage = task.stage_id
             if task and stage and not stage.allow_timesheet:
-                raise ValidationError(_(
-                    "You can't link a timesheet line to a task if its stage "
-                    "doesn't allow it. (Task: %s, Stage: %s)"
-                ) % (
+                msg = "You can't link a timesheet line to a task if its stage "
+                "doesn't allow it. (Task: {}, Stage: {})".format(
                     task.display_name,
                     stage.display_name,
```

Comment:
The error message is not displayed correctly and is cut after the word "stage".
Maybe you can also add some unit as well.